### PR TITLE
Tasks: Track token index config

### PR DESCRIPTION
**Pull request by PatricioHenderson**

### CODE REVIEW

File: src/Task.ts [hunk 1]
```diff
@@ -41,6 +43,32 @@ export function handlePriorityFeeLimitSet(event: PriorityFeeLimitSet): void {
...
+}
+
+export function handleTokensAcceptanceListSet(event: TokensAcceptanceListSet): void {
+  const task = Task.load(event.address.toHexString())
+  if (task == null) return log.warning('Missing task entity {}', [event.address.toHexString()])
+
+  const tokensAcceptanceListId = task.id
+  const tokensAcceptanceList = loadOrCreateTokenAcceptanceList(tokensAcceptanceListId)
+  const tokens = tokensAcceptanceList.tokens
+  const token = event.params.token.toHexString()
```
Comment (facuspagnuolo) [inline src/Task.ts:64]: Simply to make sure we track that token information if it wasn't tracked before:

```suggestion
  const token = loadOrCreateERC20(event.params.token).id
```

File: src/Task.ts [hunk 2]
```diff
@@ -125,6 +153,10 @@ export function getSmartVault(address: Address): Address {
   return Address.zero()
 }
 
+export function getTokensAcceptanceListId(task: Task, acceptanceType: string): string {
+  return task.id.toString() + '/' + acceptanceType
+}
+
```
Comment (facuspagnuolo) [inline src/Task.ts:159]: I think this one is not being used anymore

```suggestion
```

File: src/Task.ts [hunk 3]
```diff
@@ -41,6 +43,32 @@ export function handlePriorityFeeLimitSet(event: PriorityFeeLimitSet): void {
   task.save()
 }
 
+export function handleTokensAcceptanceTypeSet(event: TokensAcceptanceTypeSet): void {
+  const task = Task.load(event.address.toHexString())
+  if (task == null) return log.warning('Missing task entity {}', [event.address.toHexString()])
+
+  const acceptanceType = parseAcceptanceType(event.params.acceptanceType)
+  const acceptanceListId = task.id
+  const acceptanceList = loadOrCreateTokenAcceptanceList(acceptanceListId)
```
Comment (facuspagnuolo) [inline src/Task.ts:52]: It's more readable when you define a variable right before you use it:

```suggestion
  const acceptanceListId = task.id
  const acceptanceList = loadOrCreateTokenAcceptanceList(acceptanceListId)
  const acceptanceType = parseAcceptanceType(event.params.acceptanceType)
```

File: src/Task.ts (hunk 2 above)
Comment (PatricioHenderson) [reply src/Task.ts:159]: Right, forgot to delete it.

File: schema.graphql [hunk 4]
```diff
@@ -105,6 +106,13 @@ type CustomVolumeLimit @entity {
   volumeLimit: VolumeLimit!
 }
 
+type TokensAcceptanceList @entity {
+  id: ID!
+  task: Task!
+  tokensAcceptanceType: String!
```
Comment (facuspagnuolo) [inline schema.graphql:112]: What do you think if we rename it to simply `type`?

File: src/Task.ts [hunk 5]
```diff
@@ -41,6 +43,32 @@ export function handlePriorityFeeLimitSet(event: PriorityFeeLimitSet): void {
...
+  const acceptanceType = parseAcceptanceType(event.params.acceptanceType)
+  acceptanceList.tokensAcceptanceType = acceptanceType
+  acceptanceList.save()
+}
+
+export function handleTokensAcceptanceListSet(event: TokensAcceptanceListSet): void {
+  const task = Task.load(event.address.toHexString())
+  if (task == null) return log.warning('Missing task entity {}', [event.address.toHexString()])
+
+  const tokensAcceptanceListId = task.id
```
Comment (facuspagnuolo) [inline src/Task.ts:61]: To be consistent I'd rename this variable to `acceptanceList` as you didi in `handleTokensAcceptanceTypeSet`, or rename `acceptanceList` in `handleTokensAcceptanceTypeSet` to `tokensAcceptanceList`. I prefer the less verbose option tbh :)

Comment (facuspagnuolo) [reply src/Task.ts:61]: Same for the IDs variables

File: subgraph.template.yaml [hunk 6]
```diff
@@ -196,8 +196,4 @@ templates:
           handler: handleTimeLockExecutionPeriodSet
         - event: TimeLockExpirationSet(uint256)
           handler: handleTimeLockExpirationSet
-        - event: CustomVolumeLimitSet(indexed address,indexed address,uint256,uint256)
-          handler: handleCustomVolumeLimitSet
-        - event: DefaultVolumeLimitSet(indexed address,uint256,uint256)
-          handler: handleDefaultVolumeLimitSet
```
Comment (facuspagnuolo) [inline subgraph.template.yaml:202]: This shouldn't be removed. I also think the new handlers are missing here.

**Review by facuspagnuolo — APPROVED** [review]

**Pull request merged**